### ISSUE TITLE
Local variable in expression not mapped using MapExpression

### ISSUE THREAD

**Issue by jonathanmax** [AutoMapper/AutoMapper.Extensions.ExpressionMapping#79]
Models:
```c#
class ModelA
{
    public SubModelA Sub;
}
struct SubModelA
{
    public string Id;
    public static bool operator ==(SubModelA m1, SubModelA m2)
    {
        return m1.Id == m2.Id;
    }
    public static bool operator !=(SubModelA m1, SubModelA m2)
    {
        return !(m1 == m2);
    }
    public override bool Equals(object obj)
    {
        if(obj is SubModelA ma)
        {
            return this == ma;
        }
        return false;
    }
    public override int GetHashCode()
    {
        return Id.GetHashCode();
    }
}
class ModelB
{
    public SubModelB Sub;
}
struct SubModelB
{
    public string Id;
    public static bool operator ==(SubModelB m1, SubModelB m2)
    {
        return m1.Id == m2.Id;
    }
    public static bool operator !=(SubModelB m1, SubModelB m2)
    {
        return !(m1 == m2);
    }
    public override bool Equals(object obj)
    {
        if (obj is SubModelB mb)
        {
            return this == mb;
        }
        return false;
    }
    public override int GetHashCode()
    {
        return Id.GetHashCode();
    }
}
```
Configure and reproduce:
```c#
var config = new MapperConfiguration(cfg =>
{
    cfg.AddExpressionMapping();
    cfg.CreateMap<ModelA, ModelB>().ReverseMap();
    cfg.CreateMap<SubModelA, SubModelB>().ReverseMap();
});
var mapper = config.CreateMapper();

SubModelA suba = new SubModelA { Id = "s1" };

Expression<Func<ModelA, bool>> filter = m => m.Sub == suba;
var mappedfilter = mapper.MapExpression<Expression<Func<ModelB, bool>>>(filter);
```
Description: 
suba in "m => m.Sub == suba" is not mapped to type SubModelB. 
the exception is
System.InvalidOperationException: 'The binary operator Equal is not defined for the types 'SubModelB' and 'SubModelA'.'

Reproduce using Automapper 9.0.0 and AutoMapper.Extensions.ExpressionMapping 3.1.1 and .Net Core3.1 .

**Comment by BlaiseD** [comment]
The variable `suba` is a member access expression whose parent type is this `{m => (m.Sub == value(MyCurrentClass+<>c__DisplayClass3_0).suba)}`.

The following works:
```c#
            Expression<Func<ModelA, bool>> filter = m => m.Sub == new SubModelA { Id = "s1" };
            var mappedfilter = mapper.MapExpression<Expression<Func<ModelB, bool>>>(filter);
```

AutoMapper has not configured this type: `MyCurrentClass+<>c__DisplayClass3_0`.

So it's not supported now but probably fixable.   Access suba from `MyCurrentClass+<>c__DisplayClass3_0).suba` and go from there.

**Comment by BlaiseD** [comment]
Fix with PR #80 (Supporting AutoMapper v10 and fixing Issue#79).  Please try the [MyGet Build](https://myget.org/feed/automapperdev/package/nuget/AutoMapper.Extensions.ExpressionMapping).
